api/strategies.go

```diff
@@ -284,9 +284,28 @@ func (capi *census3API) launchStrategyImport(msg *api.APIdata, ctx *httprouter.H
 	// import the strategy from IPFS in background generating a queueID
 	queueID := capi.queue.Enqueue()
 	go func() {
+		internalCtx, cancel := context.WithTimeout(context.Background(), getStrategyTimeout)
+		defer cancel()
+		// check if the strategy exists in the database using the IPFS URI
 		ipfsURI := fmt.Sprintf("%s%s", capi.downloader.RemoteStorage.URIprefix(), ipfsCID)
```

Comment:
Maybe it is a good idea to add an extra check to the ipfsURI, something like https://github.com/ipfs/go-cid/blob/master/cid.go#L227.
Looking at the code seems that we assume that the CID will always be in this form: `QmWBaeu6y1zEcKbsEqCuhuDHPL3W8pZouCPdafMCRCSUWk` or empty.

And if I am not wrong at the end of this chain, they are added to the queue without any extra check and retried at least twice before being discarded.

Maybe this not in the scope of this PR but now that I saw it, I think it is worth to mention :)

Reply:
Done!